frontend/src/app/threads/[threadId]/not-found.tsx

```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import Link from "next/link";
+
+export default function NotFound() {
+  return (
+    <div className="container mx-auto px-4 py-6 h-[calc(100vh-4rem)] flex flex-col items-center justify-center">
+      <div className="text-center space-y-4">
+        <h2 className="text-2xl font-bold">404 not found.</h2>
```

Comment:
Detail.

I would just unify this with `tools/[toolName]/not-found.tsx`


There we say `404 Not Found` (here we say `404 not found.`)